src/ComputeSharp.D2D1/Shaders/Interop/Effects/PixelShaderEffect.cs

```diff
@@ -152,6 +152,15 @@ static PixelShaderEffect()
     /// </summary>
     private int numberOfInputs;
 
+    /// <summary>
+    /// The buffer with the types of inputs for the shader.
+    /// </summary>
+    /// <remarks>
+    /// This buffer is shared among effect instances, so it should not be released. It is
+    /// owned by <see cref="For{T}"/>, which will release it when the target type is unloaded.
+    /// </remarks>
+    private D2D1PixelShaderInputType* inputTypes;
```

Comment:
as above, consider combining this with the `numberOfInputs` field and just use a `ReadOnlySpan`